probe-rs/src/probe/jlink/swd.rs

```diff
@@ -771,30 +983,66 @@ impl<Probe: RawSwdIo + 'static> DAPAccess for Probe {
         address: u16,
         values: &[u32],
     ) -> Result<(), DebugProbeError> {
-        let mut transfers: Vec<SwdTransfer> = values
-            .iter()
-            .map(|v| SwdTransfer::write(port, address, *v))
-            .collect();
+        let mut succesful_transfers = 0;
 
-        perform_transfers(self, &mut transfers)?;
+        let mut idle_cycles = std::cmp::max(1, self.swd_settings().num_idle_cycles_between_writes);
 
-        for (index, result) in transfers.iter().enumerate() {
-            match &result.status {
-                TransferStatus::Ok => {}
-                TransferStatus::Failed(err) => {
-                    log::debug!(
-                        "Error in access {}/{} of block access: {}",
-                        index + 1,
-                        values.len(),
-                        err
-                    );
-                    return Err(err.clone().into());
-                }
-                TransferStatus::Pending => {
-                    // This should not happen...
-                    panic!("Error performing transfers")
+        'transfer: for _ in 0..self.swd_settings().num_retries_after_wait {
+            let mut transfers: Vec<SwdTransfer> = values
+                .iter()
+                .skip(succesful_transfers)
+                .map(|v| SwdTransfer::write(port, address, *v))
+                .collect();
+
+            let index_offset = succesful_transfers;
+
+            perform_transfers(self, &mut transfers, idle_cycles)?;
+
+            for (index, result) in transfers.iter().enumerate() {
+                match &result.status {
+                    TransferStatus::Ok => {
+                        succesful_transfers += 1;
+                    }
+                    TransferStatus::Failed(err) => {
+                        log::debug!(
+                            "Error in access {}/{} of block access: {}",
+                            index_offset + index + 1,
+                            values.len(),
+                            err
+                        );
+
+                        if err == &DapError::WaitResponse {
+                            // Clear STICKORRUN flag
+
+                            // Because we use overrun detection, we now have to clear the overrun error
```

Comment:
```suggestion
                            // Clear STICKORRUN flag.

                            // Because we use overrun detection, we now have to clear the overrun error.
```